test/performance/odp_dmafwd.c

```diff
@@ -349,6 +360,40 @@ static parse_result_t check_options(prog_config_t *config)
 				 dma_capa.compl_mode_mask);
 			return PRS_NOT_SUP;
 		}
+
+		if (odp_stash_capability(&stash_capa, ODP_STASH_TYPE_DEFAULT) < 0) {
+			ODPH_ERR("Error querying stash capabilities\n");
+			return PRS_NOK;
+		}
+
+		if ((uint32_t)config->num_thrs > stash_capa.max_stashes) {
+			ODPH_ERR("Invalid amount of stashes: %d (max: %u)\n", config->num_thrs,
+				 stash_capa.max_stashes);
+			return PRS_NOK;
+		}
+
+		if (obj_size == sizeof(uint8_t)) {
+			max_num = stash_capa.max_num.u8;
+		} else if (obj_size == sizeof(uint16_t)) {
+			max_num = stash_capa.max_num.u16;
+		} else if (obj_size <= sizeof(uint32_t)) {
+			max_num = stash_capa.max_num.u32;
+		} else if (obj_size <= sizeof(uint64_t)) {
+			max_num = stash_capa.max_num.u64;
+		} else if (obj_size <= sizeof(odp_u128_t)) {
+			max_num = stash_capa.max_num.u128;
+		} else {
+			ODPH_ERR("Invalid stash object size: %" PRIu64 "\n", obj_size);
+			return PRS_NOK;
+		}
+
+		if (config->num_inflight > max_num) {
```

Comment:
If it was scaled down, asynchronous tracking would become impossible in case a transfer is started but the ID cannot be saved for later polling. The tester would need to stop and synchronously wait for the extra transfer to finish (or alternatively leak resources) which would be quite cumbersome. If the current dimensioning becomes a problem, it can be changed later.